codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/config/ClientConfig.java

```diff
@@ -273,9 +273,7 @@ public Builder defaultDynamicValue(String rubyDefaultBlock) {
          */
         public Builder defaultValue(String value) {
             validateDefaultNotSet();
-            this.defaults = ConfigProviderChain.builder()
-                    .dynamicProvider("proc { " + value + " }")
-                    .build();
+            this.defaults = ConfigProviderChain.builder().dynamicProvider(value).build();
```

Comment:
This isn't a dynamic config value.  Use the static provider here.

Reply:
There doesn't really seem to be a difference between the two. Maybe dynamic provider should provide the proc and |cfg| wrapping.

Reply:
The most important difference is semantic - its worth conveying the intent.  But there is a behavior difference as well - the dynamic provider allows you to use a `ClientFragment`, which lets you explicitly specify a config dependency.  

I'm not sure about providing the proc and cfg wrapping - I think having the " { |cfg| .... }" is more explicit and readable.